android/src/main/java/com/swmansion/reanimated/nodes/CallFuncNode.java

```diff
@@ -18,25 +18,33 @@ public CallFuncNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
     mArgs = Utils.processIntArray(config.getArray("args"));
   }
 
-  private void beginContext() {
+  protected void beginContext() {
     mPreviousCallID = mNodesManager.updateContext.callID;
     mNodesManager.updateContext.callID = mNodesManager.updateContext.callID + '/' + String.valueOf(mNodeID);
     for (int i = 0; i < mParams.length; i++) {
       int paramId = mParams[i];
-      ParamNode paramNode = mNodesManager.findNodeById(paramId, ParamNode.class);
-      paramNode.beginContext(mArgs[i], mPreviousCallID);
+      Node paramNode = mNodesManager.findNodeById(paramId, Node.class);
+      ((ContextNode) paramNode).beginContext(mArgs[i], mPreviousCallID);
     }
   }
 
-  private void endContext() {
+  protected void endContext() {
     for (int i = 0; i < mParams.length; i++) {
       int paramId = mParams[i];
-      ParamNode paramNode = mNodesManager.findNodeById(paramId, ParamNode.class);
-      paramNode.endContext();
+      Node paramNode = mNodesManager.findNodeById(paramId, Node.class);
+      ((ContextNode) paramNode).endContext();
     }
     mNodesManager.updateContext.callID = mPreviousCallID;
   }
 
+  @Override
+  public void setValue(Object value) {
```

Comment:
Why `CallFunc` should implement `ValueManagingNode`?

Reply:
In the case a `proc` was passed as a value that needs to be set by `callback`

Reply:
If I'm not mistaken this is important for `set` too.